### ISSUE TITLE
[BUG] DataFrame crash if it contains np.int64 converted from NaT(dtype=datetime64[ns])

### ISSUE THREAD

**Issue by xxsl** [fireducks-dev/fireducks#3]
I tried to use `.astype(np.int64)` to convert a column from datetime64[ns] (which contains NaT) to int64:

```python
import numpy as np
import fireducks.pandas as pd
sample_data = [['bad', '0:01:38.77', None],
               ['aaa','0:01:32.24','0:01:38.75'],
               ['bbb','0:01:39.37','0:01:43.25'],
                ['ccc','0:01:43.25','0:01:47.00']]

df = pd.DataFrame(sample_data, columns = ["u_1", "time_1", "time_2"])

df["delta_time"] = pd.to_datetime(df["time_2"], format='%H:%M:%S.%f') - pd.to_datetime(df["time_1"], format='%H:%M:%S.%f')

# delta_time contains the time difference which should be `dtype=datetime64[ns]`
# then I want to make a new column in second by:
df["delta_time_s"] = df["delta_time"].astype(np.int64) / 1e9
```

It can be successfully executed, maybe just because of the lazy execution of FireDucks.

But from now on, any operation to this dataframe will crash:

```python
File [/usr/local/lib/python3.10/dist-packages/fireducks/core.py:235](http://172.17.0.2/usr/local/lib/python3.10/dist-packages/fireducks/core.py#line=234), in evaluate(values, options)
    233 def evaluate(values: List[fire.Value], options=None):
    234     with tracing.scope(tracing.Level.DEFAULT, "fireducks.core.evaluate"):
--> 235         ret = _evaluate(values, options)
    236     return ret

File [/usr/local/lib/python3.10/dist-packages/fireducks/core.py:220](http://172.17.0.2/usr/local/lib/python3.10/dist-packages/fireducks/core.py#line=219), in _evaluate(values, options)
    215     return fireducks_ext.execute(
    216         context().ext_context, options.compile_options, fi
    217     )
    219 try:
--> 220     return fire.evaluate(values, wrapper, package="fireducks")
    221 except fireducks_ext.InvalidIndexError as e:
    222     raise pandas.errors.InvalidIndexError(e)

File [/usr/local/lib/python3.10/dist-packages/firefw/runtime.py:31](http://172.17.0.2/usr/local/lib/python3.10/dist-packages/firefw/runtime.py#line=30), in prohibit_reentrant.<locals>.wrapper(*args, **kwargs)
     29 lock = True
     30 try:
---> 31     ret = func(*args, **kwargs)
     32 finally:
     33     lock = False

File [/usr/local/lib/python3.10/dist-packages/firefw/runtime.py:72](http://172.17.0.2/usr/local/lib/python3.10/dist-packages/firefw/runtime.py#line=71), in evaluate(values, executor, package)
     67 with tracing.scope(tracing.Level.VERBOSE, "create_mlir_func"):
     68     source, input_values, output_values = create_mlir_func(
     69         values, package=package
     70     )
---> 72 ret = executor(source, input_values, output_values)
     74 if len(output_values) != len(ret):
     75     raise RuntimeError("Executor did not return enough number of returns")

File [/usr/local/lib/python3.10/dist-packages/fireducks/core.py:215](http://172.17.0.2/usr/local/lib/python3.10/dist-packages/fireducks/core.py#line=214), in _evaluate.<locals>.wrapper(ir, input_values, output_values)
    213 fi.input_values = [v.get_result() for v in input_values]
    214 fi.output_types = [v.mlir_type for v in output_values]
--> 215 return fireducks_ext.execute(
    216     context().ext_context, options.compile_options, fi
    217 )

RuntimeError: Integer value -9223372036854775808 not in range: -9007199254740992 to 9007199254740992
```

![2024-02-12_15-33](https://github.com/fireducks-dev/fireducks/assets/7067381/e7021be5-68ed-495d-a9c0-6ace6519ac02)

Even the operation that not related to the new int64 column crashed:

![2024-02-12_15-37](https://github.com/fireducks-dev/fireducks/assets/7067381/c5f0bf7b-ba30-42cb-be8c-e87a446158f7)

On the contrary, original Pandas works well on this case:

![2024-02-12_15-33_1](https://github.com/fireducks-dev/fireducks/assets/7067381/03fc5d0d-8772-494a-8100-bb88a4373abb)

Please confirm this, tks

- FireDucks version: 0.9.8
- OS: Ubuntu 22.04.3 LTS (GNU/Linux 6.1.0-17-amd64 x86_64)
- Python3 version: Python 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0] on linux

**Comment by qsourav** [comment]
Thank you for reporting the issue so elaborately. 

We have fixed the same in our latest release: 0.10.0
https://fireducks-dev.github.io/docs/release-note/#0.10.0

We would be glad if you like to evaluate your program with this version and confirm the situation.

**Comment by xxsl** [comment]
Hi @qsourav,

Thanks for the reply.
I have tested the case on FireDucks 0.10.0 and everything goes well now.

![2024-02-14_15-58](https://github.com/fireducks-dev/fireducks/assets/7067381/bbf64eda-b6bf-4f17-9402-9f8716f7ed21)

So surprised you guys get things done so quick ;-)
